common/logger/src/struct_log.rs

```diff
@@ -94,13 +94,18 @@ impl Write for TcpWriter {
         // Attempt to write, and if it fails clear underlying stream
         // This doesn't guarantee a message cut off mid send will work, but it does guarantee that
         // we will connect first
-        let mut stream = self.stream.as_ref().unwrap();
-        let result = stream.write(buf);
-        if result.is_err() {
-            self.stream = None;
+        if let Some(mut stream) = self.stream.as_ref() {
+            let result = stream.write(buf);
+            if result.is_err() {
+                self.stream = None;
+            }
+            result
+        } else {
+            Err(io::Error::new(
+                io::ErrorKind::NotConnected,
+                "Can't write to stream, not connected",
+            ))
         }
```

Comment:
I think this would be more cleanly written as a set of Rust functors:

```
self.stream
  .as_mut()
  .ok_or_else(|| io::Error::new(...))
  .and_then(|stream| stream.write(...).map_err(|e| { self.stream = None; e })
```

Reply:
that's the annoying things with rust, you can write everything in imperative or functional style, so which do you choose :P?